### PR TITLE
[FAI-13484] Custom component for dashboard summarizer

### PR DESCRIPTION
**Pull request by fedetoledo**
> [!IMPORTANT]
> If you are merging into master, please add either a `backport` or a `no-backport` label to this PR. You will not be able to merge until you do this step. Refer to the section [Do I need to backport this PR?](https://www.notion.so/metabase/Metabase-Branching-Strategy-6eb577d5f61142aa960a626d6bbdfeb3?pvs=4#89f80d6f17714a0198aeb66c0efd1b71) in the Metabase Branching Strategy document for more details.

> **Warning**
>
> If that is your first contribution to Metabase, please sign the [Contributor License Agreement](https://docs.google.com/a/metabase.com/forms/d/1oV38o7b9ONFSwuzwmERRMi9SYrhYeOrkbmNaq9pOJ_E/viewform) (unless it's a tiny documentation change). Also, if you're attempting to fix a translation issue, please submit your changes to our [POEditor project](https://poeditor.com/join/project/ynjQmwSsGh) instead of opening a PR.

Closes https://github.com/metabase/metabase/issues/[issue_number]

### Description

Describe the overall approach and the problem being solved.

### How to verify

Describe the steps to verify that the changes are working as expected.

1. New question -> Sample Dataset -> ...
2. ...

### Demo

_Upload a demo video or before/after screenshots if sensible or remove the section_

### Checklist

- [ ] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase-types/store/dashboard.ts [hunk 1]
```diff
@@ -55,12 +55,17 @@ export type DashboardSummary = {
   error?: string;
 };
 
+export type CustomDashboardSummary = {
+  summary: string;
```
Comment (kinfaikan) [inline frontend/src/metabase-types/store/dashboard.ts:59]: is this correct?

File: frontend/src/metabase/dashboard/components/DashboardLighthouseAISidebar/DashboardLighthouseAISidebar.tsx [hunk 2]
```diff
@@ -46,76 +39,11 @@ interface CopyToClipboardProps {
 const LOADING = "Loading dashboard ...";
 const FAILED =
   "We encountered an error while running the analysis. Please try again later.";
-const RUNNING = "Analyzing dashboad ... Please wait, this may take a moment.";
+const RUNNING = "Analyzing dashboard ... Please wait, this may take a moment.";
```
Comment (kinfaikan) [inline frontend/src/metabase/dashboard/components/DashboardLighthouseAISidebar/DashboardLighthouseAISidebar.tsx:42]: nice catch!

File: frontend/src/metabase/redux/embed.ts [hunk 3]
```diff
@@ -16,6 +16,7 @@ export const DEFAULT_EMBED_OPTIONS = {
   action_buttons: true,
   enable_chart_explainer: false,
   enable_dashboard_creation: false,
+  enable_dashboard_summarizer: true,
```
Comment (kinfaikan) [inline frontend/src/metabase/redux/embed.ts:19]: default to true?

File: frontend/src/metabase/visualizations/visualizations/LighthouseSummary/LighthouseSummary.jsx [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+          />
+          <div
+            style={{
+              color: "#072E45",
+              fontWeight: 600,
+              fontSize: "19px",
+              marginTop: "16px",
+            }}
+          >
+            Summarize Unavailable
```
Comment (kinfaikan) [inline frontend/src/metabase/visualizations/visualizations/LighthouseSummary/LighthouseSummary.jsx:190]: typo

File: frontend/src/metabase/visualizations/visualizations/LighthouseSummary/LighthouseSummary.jsx [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+          </div>
+          <div style={{ marginTop: "8px", textWrap: "balance" }}>
+            All charts have no data or contain errors in their query
```
Comment (kinfaikan) [inline frontend/src/metabase/visualizations/visualizations/LighthouseSummary/LighthouseSummary.jsx:193]: in general, all targeted charts are not necessarily the same as all charts.

File: frontend/src/metabase-types/store/dashboard.ts [hunk 6]
```diff
@@ -55,12 +55,19 @@ export type DashboardSummary = {
   error?: string;
 };
 
+export type CustomDashboardSummary = {
```
Comment (kinfaikan) [inline frontend/src/metabase-types/store/dashboard.ts:58]: nit: it is identical to `DashboardSummary`.

File: frontend/src/metabase/dashboard/actions/cards.js [hunk 7]
```diff
@@ -184,6 +184,29 @@ export const addHeadingDashCardToDashboard = function ({ dashId, tabId }) {
   });
 };
 
+export const addLighthouseSummaryToDashboard = function ({ dashId, tabId }) {
+  trackCardCreated("lighthouseSummary", dashId);
+
+  const virtualTextCard = {
+    ...createCard(),
+    display: "lighthouseSummary",
+    archived: false,
```
Comment (kinfaikan) [inline frontend/src/metabase/dashboard/actions/cards.js:193]: just curious, do you know what `archived` is for?

File: frontend/src/metabase/dashboard/components/DashboardLighthouseAISidebar/useDashboardSummarizer.tsx [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+    dashboardSummary,
+  ]);
+
+  useEffect(() => {
+    if (!isEditing) {
+      postMessage();
+    }
+  }, [isEditing, postMessage]);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (kinfaikan) [inline frontend/src/metabase/dashboard/components/DashboardLighthouseAISidebar/useDashboardSummarizer.tsx:72]: nit: this comment line seems redundant?

File: frontend/src/metabase/visualizations/visualizations/LighthouseSummary/LighthouseSummary.jsx [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+            Add Charts to Summarize
+          </div>
+          <div style={{ marginTop: "8px", textWrap: "balance" }}>
+            This tool summarizes charts in your dashboard. Please add charts
+            before using this feature.
+          </div>
+        </CenteredContainer>
+      )}
+
+      {statusMessage === Status.NO_QUERY_RESULTS_FOR_ALL_TARGETED_CHARTS && (
```
Comment (kinfaikan) [inline frontend/src/metabase/visualizations/visualizations/LighthouseSummary/LighthouseSummary.jsx:170]: when no heuristics are available for a dashboard, the custom dashboard summarizer falls back to the regular version and may return `NO_QUERY_RESULTS_FOR_ALL_CHARTS`.

**Review by kinfaikan — APPROVED** [review]

File: frontend/src/metabase/dashboard/actions/cards.js (hunk 7 above)
Comment (fedetoledo) [reply frontend/src/metabase/dashboard/actions/cards.js:193]: Not really, just replicated the text component

File: frontend/src/metabase/dashboard/components/DashboardLighthouseAISidebar/useDashboardSummarizer.tsx (hunk 8 above)
Comment (fedetoledo) [reply frontend/src/metabase/dashboard/components/DashboardLighthouseAISidebar/useDashboardSummarizer.tsx:72]: it's for the next occurrence, throwing a warning

**Pull request merged**